slide_analysis/initialize_base_service/init_base_service.py

```diff
@@ -0,0 +1,30 @@
+import os
+import pickle
+
+
+class InitBaseService:
+    def __init__(self, path):
+        if path[-1] == '/':
+            self.base_path = path
+        else:
+            self.base_path = path + '/'
+        if not os.path.exists(self.base_path):
+            os.makedirs(self.base_path)
+
+    def write_to_file(self, tile):
+        descr = self.descriptor_class(self.descriptor_params)
+        descr.calc_by_tile(tile)
+        pickle.dump(descr, self.opened_descr_file)
+
+    def init_base(self, tile_stream, descriptor_class, descriptor_params, imagename, classname):
```

Comment:
what is classname variable? Which class name are you expecting in params?

Reply:
Why shouldn't we move all the initialization to `__init__` and we have only method `store` which only stores the data to file?

Reply:
We will have one instance of InitBaseService but use it many times, I mean if we need to create two bases of descriptors (e.g. different types or configurations) then this instance will store both of them. So as result initalizing of service with specific data is pointless

Reply:
Except tile_stream, descriptor_class, params and all other func parameters you store only `path` variable. So what is the point of reusing this class if as we see there is definetly no winning in memory and I believe in performance? But we lose in clearence of code as we can have constructor with all theese parameters followed by some `run` method with no parameters.This allowes us to generate an instance of this class and forward it to some other function which can call for `run` method without any additional parameters which is clearly better for modularity of the project

Reply:
Of course that`s right idea, but only in case when application is able to create database of descriptors only one time. In our case we may have need to create many databases while application is running. So using your solution we will create new instance of InitBaseService for calling one function each time we need to create new database, but I think it will be better to have one fixed instance of it.

Reply:
Yep that's exactly what i mean one instance for each database creation. So what is the problem with initializating a new instance for another db creation? You can create hundreds of instances and run them. This is more flexible than having only one and use it several times. Not to mention the amount of time you need to create a database. You have to be very patient to create at least one